File: packages/dashboard/src/components/schedule-visualizer/tests/building-map.ts
import * as RmfModels from 'rmf-models';
import airportMap from './building-map-airport';
import officeMap from './building-map-office';

export default async function buildingMap(): Promise<RmfModels.BuildingMap> {
  return {
    name: 'test building',
    levels: [
      {
        name: 'L1',
        elevation: 0,
        images: [
          {
            data: '',
            encoding: 'png',
            name: 'office',
            scale: 0.008465494960546494,
            x_offset: 0,
            y_offset: 0,
            yaw: 0,
          },
        ],
        doors: officeMap.doors,
        places: officeMap.places,
        nav_graphs: officeMap.nav_graphs,
        wall_graph: {
          name: 'wallgraph',
          vertices: [],
          edges: [],
          params: [],
        },
      },
      {
        name: 'Airport Terminal',
        elevation: 10,
        images: [
          {
            data: '',
            encoding: 'png',
            name: 'airport_terminal',
            scale: 0.08212187141180038,
            x_offset: 0,
            y_offset: 0,
            yaw: 0,
          },
        ],
        doors: airportMap.doors,
        places: airportMap.places,
        nav_graphs: [],
        wall_graph: {
          name: 'wallgraph',
          vertices: [],
          edges: [],
          params: [],
        },
      },
    ],
    lifts: officeMap.lifts,
  };
}

File: packages/api-server/api_server/rmf_io/test_data/building_map.py
import os.path

from building_map_msgs.msg import AffineImage, BuildingMap, Graph, Level


def make_building_map():
    with open(f"{os.path.dirname(__file__)}/office.png", "br") as f:
        image_data = f.read()

    return BuildingMap(
        name="test_name",
        levels=[
            Level(
                name="L1",
                elevation=0.0,
                images=[
                    AffineImage(
                        name="test_image",
                        x_offset=0.0,
                        y_offset=0.0,
                        yaw=0.0,
                        scale=1.0,
                        encoding="png",
                        data=image_data,
                    )
                ],
                places=[],
                doors=[],
                nav_graphs=[],
                wall_graph=Graph(
                    name="test_graph",
                    vertices=[],
                    edges=[],
                    params=[],
                ),
            ),
        ],
        lifts=[],
    )
